### PR TITLE
[MNT] Added post-merge hook and updated scripts

### LINKED ISSUE #43: [MNT] : Update Hooks and Commands for Dependency Management and Code Quality
### Is there an existing issue for this?

- [X] I have searched the existing issues

### New feature

This issue addresses the implementation of several enhancements aimed at improving dependency management and code quality within the project. The changes proposed in this pull request include the addition of a post-merge hook for automatic dependency updates, support for type checking in the pre-commit hook, and the inclusion of commands for fixing ESLint and formatting with Prettier.

`Proposed Changes`
**Post-Merge Hook Addition**

Added a post-merge hook to automate dependency updates following changes to the package.json file. This aligns with best practices commonly observed in JavaScript projects, ensuring that dependencies remain up-to-date.

**Type Check Support in Pre-Commit Hook**

Included support for type checking within the pre-commit hook. This addition enhances code quality by catching type-related errors before they are committed to the repository, thus reducing the likelihood of introducing bugs.

**ESLint Problem Fixing Command.**

Added the command "lint:fix": "eslint --fix **/*.{ts,tsx}" to address ESLint issues automatically. This command facilitates adherence to coding standards and improves code consistency by automatically fixing ESLint problems.
Prettier Format Checking Command.

Introduced the command "format:check": "prettier --check ." to verify code formatting consistency using Prettier. This command ensures that code adheres to predefined formatting rules, promoting readability and maintainability across the project.

Objective
The objective of these changes is to streamline development processes, enhance code quality, and maintain consistency within the project. By automating dependency updates, enforcing type checking, and providing tools for fixing ESLint issues and verifying code formatting, this initiative aims to improve overall project health and developer efficiency.

**Tasks**

- [x] Review and validate the proposed changes in the pull request.
- [x] Ensure compatibility and alignment with project requirements and coding standards.
- [x] Test the effectiveness of the added commands and hooks in improving development workflows.
- [ ] Update documentation to reflect the changes introduced by this enhancement.
 Merge the pull request after thorough review and testing.
 
**Related Issue**
#11 : These enhancements are based on established best practices and aim to address common pain points in JavaScript development workflows. The inclusion of automated processes for dependency management and code quality checks aligns with industry standards and contributes to a more robust and maintainable codebase.

### Unclear documentation

_No response_

### PR DESCRIPTION
**Pull request by Sauradip07**
- fixes #43 
Changes proposed in this pull request:

- Added post-merge hook: This is a common practice in many JavaScript projects to automatically update dependencies after changes to the package.json file.

- Added type check support in pre-commit hook.

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Review by rmanaem — COMMENTED** [review]
Thanks for the PR @Sauradip07!
Please open an issue and describe why the proposed changes are required/beneficial.
We can then proceed with the review once the issue is linked here.

**Comment by Sauradip07** [comment]
> Thanks for the PR @Sauradip07! Please open an issue and describe why the proposed changes are required/beneficial. We can then proceed with the review once the issue is linked here.

Please Check

File: .husky/post-merge [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+git diff HEAD^ HEAD --exit-code -- ./package.json || npm install
```
Comment (rmanaem) [inline .husky/post-merge]: I think a clean install would be better here, see https://docs.npmjs.com/cli/v10/commands/npm-ci

File: .husky/pre-commit [hunk 2]
```diff
@@ -2,3 +2,5 @@
 . "$(dirname -- "$0")/_/husky.sh"
 
 npx lint-staged
+npm run typecheck
```
Comment (rmanaem) [inline .husky/pre-commit]: As mentioned in our discussion on #43 I think type check is unnecessary here as we have configured ESLint with appropriate plugins and rules to take care of this.

File: package.json [hunk 3]
```diff
@@ -6,9 +6,12 @@
   "scripts": {
     "dev": "vite",
     "build": "tsc && vite build",
-    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives",
+    "lint:check": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings=0",
+    "lint:fix": "eslint --fix **/*.{ts,tsx}",
     "preview": "vite preview",
-    "format": "prettier --write .",
+    "format:fix": "prettier --write .",
+    "format:check": "prettier --check .",
```
Comment (rmanaem) [inline package.json:13]: Nice!

File: package.json [hunk 4]
```diff
@@ -6,9 +6,12 @@
...
+    "typecheck": "tsc --project tsconfig.json --noEmit",
```
Comment (rmanaem) [inline package.json]: We can also remove this.

File: .lintstagedrc.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-  "*.{ts,tsx}": ["npm run lint", "npm run format"]
+  "*.{ts,tsx}": ["npm run lint:check", "npm run format:fix"]
```
Comment (rmanaem) [inline .lintstagedrc.json]: Let's use the `format:check` for `prettier` here as well.

**Review by rmanaem — CHANGES_REQUESTED** [review]
Thanks for drafting the issue!
I've requested some changes following our discussion on #43, please take a look.

File: .lintstagedrc.json [hunk 6]
```diff
@@ -1,3 +1,3 @@
 {
-  "*.{ts,tsx}": ["npm run lint", "npm run format"]
+  "*.{ts,tsx}": ["npm run lint:check", "npm run format:fix", "npm run format:check"]
```
Comment (rmanaem) [inline .lintstagedrc.json]: I think we can remove the `npm run format:fix` now that we have the `npm run format:check`.

**Review by rmanaem — CHANGES_REQUESTED** [review]
Thanks for implementing changes!
There is one more thing left and after that this PR should be ready to merge.
I'd also suggest updating the title to something like `[MNT] Added post-merge hook and updated scripts` since the PR no longer includes integration of type check support.

**Review by rmanaem — APPROVED** [review]
I believe this is ready to merge 🧑‍🍳

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/neurobagel/query-tool/releases/tag/v0.1.0) :rocket:

**Pull request merged**